src/Umbraco.Web.BackOffice/Trees/FilesTreeController.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Umbraco.Core;
+using Umbraco.Core.Hosting;
+using Umbraco.Core.IO;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Services;
+using Umbraco.Web.BackOffice.Trees;
+using Umbraco.Web.WebApi;
+
+namespace Umbraco.Web.Trees
+{
+    [Tree(Constants.Applications.Settings, "files", TreeTitle = "Files", TreeUse = TreeUse.Dialog)]
+    [CoreTree]
+    public class FilesTreeController : FileSystemTreeController
+    {
+        protected override IFileSystem FileSystem { get; }
+
+        private static readonly string[] ExtensionsStatic = { "*" };
+
+        public FilesTreeController(
+            ILocalizedTextService localizedTextService,
+            UmbracoApiControllerTypeCollection umbracoApiControllerTypeCollection,
+            IMenuItemCollectionFactory menuItemCollectionFactory,
+            IIOHelper ioHelper,
+            IHostingEnvironment hostingEnvironment,
+            ILogger logger)
+            : base(localizedTextService, umbracoApiControllerTypeCollection, menuItemCollectionFactory)
+        {
+            FileSystem = new PhysicalFileSystem(ioHelper, hostingEnvironment, logger, "~/");
```

Comment:
I imagine there's a reason for this, but can we inject an IFileSystem instead of newing up a physical one in the contructor?

Reply:
No good reason, other then it was the same way in V8. I update to inject a specific IFileSystem, that is setup in `BackOfficeComposer`